exporters/otlp/otlpmetric/otlpmetrichttp/exporter_test.go

```diff
@@ -0,0 +1,124 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otlpmetrichttp // import "go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/internal/oconf"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/internal/otest"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+)
+
+func TestExporterClientConcurrentSafe(t *testing.T) {
+	const goroutines = 5
+
+	coll, err := otest.NewHTTPCollector("", nil)
+	require.NoError(t, err)
+
+	ctx := context.Background()
+	addr := coll.Addr().String()
+	opts := []Option{WithEndpoint(addr), WithInsecure()}
+	cfg := oconf.NewHTTPConfig(asHTTPOptions(opts)...)
+	client, err := newClient(cfg)
+	require.NoError(t, err)
+
+	exp, err := newExporter(client, oconf.Config{})
+	require.NoError(t, err)
+	rm := new(metricdata.ResourceMetrics)
+
+	done := make(chan struct{})
+	first := make(chan struct{}, goroutines)
```

Comment:
I think it would easier and more consistent to create another `sync.WaitGroup`.

Reply:
This test it copied from `otlpmetric/internal`. I don't plan to change it in this PR to avoid confusion. Please feel free to address it in a follow up PR if you feel it needs to be changed.

Reply:
https://github.com/open-telemetry/opentelemetry-go/issues/4399